addons/stock/static/src/js/report_stock_reception.js

```diff
@@ -0,0 +1,214 @@
+        if (! ev.currentTarget.className.includes('o_assign_all')) {
+            nodeToAssign = [ev.currentTarget];
+            ev.target.closest('tbody').previousElementSibling.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
+        } else {
+            ev.target.style.display = 'none';
+            if (ev.target.name === "assign_all_link") {
+                this.iframe.contentDocument.querySelectorAll('.o_assign_all').forEach(b => b.style.display = 'none');
+                this.iframe.contentDocument.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
+                nodeToAssign = this.iframe.contentDocument.querySelectorAll('.o_report_reception_assign:not(.o_assign_all)');
+            } else {
+                // Local assign all
+                nodeToAssign = ev.target.closest('thead').nextElementSibling.querySelectorAll('.o_report_reception_assign:not(.o_assign_all)');
+                ev.target.closest('thead').nextElementSibling.querySelectorAll('.o_print_label_all').forEach(button => button.removeAttribute('disabled'));
+            }
+        }
+        nodeToAssign.forEach(node => {
+            node.closest('td').nextElementSibling.querySelectorAll('.o_print_label').forEach(button => button.removeAttribute('disabled'));
+            moveIds.push(parseInt(node.getAttribute('move-id')));
+            quantities.push(parseFloat(node.getAttribute('qty')));
+            inIds.push(JSON.parse(node.getAttribute('move-ins-ids')));
+            this._switchButton(node);
+        });
+
+        return this._rpc({
+            model: 'report.stock.report_reception',
+            args: [false, moveIds, quantities, inIds],
+            method: 'action_assign'
+        })
+    },
+
+    /**
+     * Unassign the specified move
+     *
+     * @returns {Promise}
+     */
+     _onClickUnassign: function(ev) {
+        this._switchButton(ev.currentTarget);
+        let quantity = parseFloat(ev.target.getAttribute('qty'));
+        let modelId = parseInt(ev.target.getAttribute('move-id'));
+        let inIds = JSON.parse("[" + ev.target.getAttribute('move-ins-ids') + "]");
+        ev.target.closest('td').nextElementSibling.querySelectorAll('.o_print_label').forEach(button => button.setAttribute('disabled', true));
+        return this._rpc({
+            model: 'report.stock.report_reception',
+            args: [false, modelId, quantity, inIds[0]],
+            method: 'action_unassign'
+        })
+    },
+
+    /**
+     * Open the forecast report for the product of
+     * the selected move
+     */
+    _onClickForecastReport: function(ev) {
+        const model = ev.target.getAttribute('model');
+        const modelId = parseInt(ev.target.getAttribute('move-id'));
+        return this._rpc( {
+            model,
+            args: [[modelId]],
+            method: 'action_product_forecast_report'
+        }).then((action) => {
```

Comment:
Is the model can be something else than a move ? If not, I would remove the attribute from the xml template and pass `stock.move` directly here as following:
```suggestion
        const modelId = parseInt(ev.target.getAttribute('move-id'));
        return this._rpc({
            model: 'stock.move',
            args: [[modelId]],
            method: 'action_product_forecast_report'
        }).then(action => {
```